Frigate Camera Notifications/Beta

```diff
@@ -809,212 +881,336 @@ variables:
-                              push:
-                                sound:
-                                  name: "{{sound}}"
-                                  volume: "{{ iif(sound == 'none', 0, volume) }}"
-                                  critical: "{{ iif(critical, 1, 0) }}"
-                              entity_id: "{{ios_live_view}}"
-                              # Actions
-                              actions:
-                                - action: URI
-                                  title: "{{button_1}}"
-                                  uri: "{{url_1}}"
-                                  icon: "{{icon_1}}"
-                                - action: URI
-                                  title: "{{button_2}}"
-                                  uri: "{{url_2}}"
-                                  icon: "{{icon_2}}"
-                                - action: "{{ 'URI' if '/' in url_3 else url_3 }}"
-                                  title: "{{button_3}}"
-                                  uri: "{{url_3}}"
-                                  icon: "{{icon_3}}"
-                                  destructive: true
-                      - conditions: "{{ tv }}"
-                        sequence:
-                          - service: "notify.{{ notify_group_target }}"
+  redacted: !input redacted
+  master_condition: !input master_condition
+  # Dynamic Variables per event
+  objects: "{{ trigger.payload_json['after']['data']['objects'] if trigger.id == 'frigate-event' }}"
+  initial_home: "{{ presence_entity != '' and is_state(presence_entity, 'home') }}"
+  after_zones: "{{ trigger.payload_json['after']['data']['zones'] if trigger.id == 'frigate-event'}}"
+  zone_multi_filter: "{{zone_only and zone_multi and after_zones|length and zones and zones |reject('in', after_zones) |list |length == 0 }}"
+conditions:
+  condition: or
+  conditions:
+    - condition: trigger
+      id: silence
+    - condition: trigger
+      id: custom
+    - condition: and
+      conditions:
+        - condition: trigger
+          id: frigate-event
+        - "{{ input_camera_name|select('equalto', camera)|list|length>0 }}"
+        - alias: "Notifications enabled for object"
+          condition: template
+          value_template: "{{ not labels|length or labels|select('in', objects)|list|length > 0 }}"
+        - "{{ not this.attributes.last_triggered or (now() - this.attributes.last_triggered).seconds > cooldown }}"
+        - "{{ not disable_times|length or not now().hour in disable_times|map('int')|list }}"
+        - "{{ not zone_only or (not zone_multi and zones|select('in', after_zones)|list|length > 0) or (zone_multi and after_zones|length > 0 and zones |reject('in', after_zones) |list |length == 0) }}"
+        - "{{ not initial_home }}"
+        - "{{ not state_only or states(input_entity) in states_filter }}"
+        - "{{ custom_filter }}"
+        - "{{ type == 'new' or type  == 'end' or update_thumbnail }}"
```

Comment:
I'm finding that I only get alerts during 'end' messages rather than during events and this looks to be the culprit. My experience so far, and from what I've read from the Frigate devs, 'update' types can assign zones as well. When zone filter is off, this should be fine, but when it's on, allow update types where the zones change (before.data.zones is different than after.data.zones).

Reply:
Okay good feedback. I don't use zone filters much at all now with the way reviews work so appreciate the info. 

This was designed to stop you receiving dozens of updates for one event but i agree with your point. 

You think the solution would be to test against the state of the zone filters as well?

E.g type  == 'update' if zone_filter

Reply:
Makes sense. That would work for my usage, but not sure if there are other updates (e.g., types of objects) that people would care about. Those are the two I can think of.

I'm going to largely eliminate zone usage as well, but there are a couple of circumstances I want separate notifications (e.g. stopping at my mailbox vs entering my driveway) and don't want the cool down to prevent sending a notification.

Reply:
Here's my initial take:

```
  zone_updated: '{{ trigger.payload_json[''before''][''data''][''zones''] != trigger.payload_json[''after''][''data''][''zones''] }}'
  object_updated: '{{ trigger.payload_json[''before''][''data''][''objects''] != trigger.payload_json[''after''][''data''][''objects''] }}'
  sublabel_updated: '{{ trigger.payload_json[''before''][''data''][''sub_labels''] != trigger.payload_json[''after''][''data''][''sub_labels''] }}'
}}'

```
then criteria:

    - '{{ type == ''new'' or update_thumbnail or zone_updated or object_updated or (sublabel_updated and not alert_once) }}'

I've done a couple of tests and so far it seems to work as expected. I can't think of a reason that type == "end" would be needed here as it would be otherwise covered by zone, object or sublabel updates as needed.

The caveat here is that if zones or objects flap, it could result in additional alerts.

Reply:
type = end is needed for IOS to get the video

Reply:
> Here's my initial take:
> 
> ```
>   zone_updated: '{{ trigger.payload_json[''before''][''data''][''zones''] != trigger.payload_json[''after''][''data''][''zones''] }}'
>   object_updated: '{{ trigger.payload_json[''before''][''data''][''objects''] != trigger.payload_json[''after''][''data''][''objects''] }}'
>   sublabel_updated: '{{ trigger.payload_json[''before''][''data''][''sub_labels''] != trigger.payload_json[''after''][''data''][''sub_labels''] }}'
> }}'
> ```
> 
> then criteria:
> 
> ```
> - '{{ type == ''new'' or update_thumbnail or zone_updated or object_updated or (sublabel_updated and not alert_once) }}'
> ```
> 
> I've done a couple of tests and so far it seems to work as expected. I can't think of a reason that type == "end" would be needed here as it would be otherwise covered by zone, object or sublabel updates as needed.
> 
> The caveat here is that if zones or objects flap, it could result in additional alerts.

I like this however there are a lot more cases we need to cover. 

sublabel only matters if person is in the input labels list for example
zones only matter if zone filter is enabled AND the zones that change are relevant

Reply:
```
  before_zones: "{{ trigger.payload_json['before']['data']['zones'] if trigger.id == 'frigate-event' }}"
  after_zones: "{{ trigger.payload_json['after']['data']['zones'] if trigger.id == 'frigate-event' }}"
  zone_multi_filter: "{{zone_only and zone_multi and after_zones|length and zones and zones |reject('in', after_zones) |list |length == 0 }}"
  # still need to add somemore conditions to these to catch all use cases, especially the zones.  
  zone_updated: >
                  {% if trigger.id == 'frigate-event' %} 

                    {{ before_zones | select('in', zones) | list | length != after_zones| select('in', zones) | list | length }}
                  {% endif %}
  object_updated: >
                  {% if trigger.id == 'frigate-event' %}
                    {{ trigger.payload_json['before']['data']['objects'] | select('in', objects) | list | length  != objetcs | select('in', objects) | list | length }}
                  {% endif %}
  sublabel_updated: >
                  {% if trigger.id == 'frigate-event' and 'person' in input_labels %}
                    {{trigger.payload_json['before']['data']['sub_labels'] != trigger.payload_json['after']['data']['sub_labels'] }}
                  {% endif %}
```